### PR TITLE
cache chainID in client

### PR DESCRIPTION
**Pull request by wangdayong228**
This change is [Reviewable](https://reviewable.io/reviews/conflux-chain/go-conflux-sdk/134)

### CODE REVIEW

File: client.go [hunk 1]
```diff
@@ -247,32 +255,50 @@ func (client *Client) GetStatus() (status types.Status, err error) {
 
 // GetNetworkID returns networkID of connecting conflux node
 func (client *Client) GetNetworkID() (uint32, error) {
-	if client.networkID != 0 {
-		return client.networkID, nil
+	if client.networkID != nil {
```
Comment (boqiu) [inline client.go:258]: Basically, pointer type is ok. However, `atomic.Value` is  better.

File: client.go [hunk 2]
```diff
@@ -247,32 +255,50 @@ func (client *Client) GetStatus() (status types.Status, err error) {
...
 	}
 
 	status, err := client.GetStatus()
 	if err != nil {
 		return 0, errors.Wrap(err, "failed to get status")
 	}
 
-	client.networkID = uint32(status.NetworkID)
-	return client.networkID, nil
+	networkID := uint32(status.NetworkID)
```
Comment (boqiu) [inline client.go:267]: Could init chainId at the same time.

**Review by boqiu — CHANGES_REQUESTED** [review]

**Comment by wangdayong228** [comment]
add methods for obtaining cached networkID and chainID

**Comment by wangdayong228** [comment]
add methods for obtaining cached networkID and chainID

**Comment by wangdayong228** [comment]
add methods for obtaining cached networkID and chainID

**Pull request merged**